### PR TITLE
(DOCSP-12223): Link User Identities, Android

### PR DESCRIPTION
**Pull request by nathan-contino-mongo**
## Pull Request Info
I would like to include some information about what happens to user ids, custom user data, etc. But I wanted to check first to make sure that's not going too far.

### Issue JIRA link:
https://jira.mongodb.org/browse/DOCSP-12223

### Docs staging link (requires sign-in on MongoDB Corp SSO):
https://docs-mongodbcom-staging.corp.mongodb.com/realm/pegasus/DOCSP-12223/android/link-identities.html

### CODE REVIEW

File: source/android/link-identities.txt [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+multiple identities corresponding to a single user into one user
+identity.
+
+Example
+-------
+
+For instance, consider an application that enables users to authenticate
+via :ref:`anonymous authentication <anonymous-authentication>` so they
+may explore functionality without the hassle of linking to an existing
+SSO provider or generating a new password. When a user decides to
```
Comment (ccho-mongodb) [inline source/android/link-identities.txt:22]: Nitpick: I think "When that user" enforces this is an antecedent of the hypothetical situation.

File: source/android/link-identities.txt [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+finally create a full account with an SSO provider or email/password
+authentication, you need some way of persisting the user's original
+anonymous identity with their new permanent identity.
+
+You can link identities using the ``linkCredentials()`` or
+``linkCredentialsAsync()`` methods of the ``User`` object of a logged in
+User.
```
Comment (ccho-mongodb) [inline source/android/link-identities.txt:29]: Nitpick: should "User" be preformatted uppercase or unformatted lowercase?

File: source/android/link-identities.txt [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+            }
+         }
+
+   .. tab::
+      :tabid: java
+
+      .. code-block:: java
+
+         public void linkUserWithEmailPasswordUser(User user, String email, String password) {
+            user.linkCredentialsAsync(Credentials.emailPassword(email, password), it -> {
```
Comment (ccho-mongodb) [inline source/android/link-identities.txt:54]: Question: is `it` idiomatic in Android for anonymous function parameters? I see `it` is a keyword in Kotlin, but if not idiomatic for Android, maybe `result` could be more descriptive and helpful.

File: source/android/link-identities.txt [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+corresponds to a unique user identity, {+service-short+} lets you merge
+multiple identities corresponding to a single user into one user
+identity.
+
+Example
+-------
+
+For instance, consider an application that enables users to authenticate
+via :ref:`anonymous authentication <anonymous-authentication>` so they
+may explore functionality without the hassle of linking to an existing
```
Comment (ccho-mongodb) [inline source/android/link-identities.txt:21]: Nitpick: would omit "the hassle of" since it is editorial.

**Review by ccho-mongodb — APPROVED** [review]
LGTM, a couple nitpicks.

File: source/android/link-identities.txt (hunk 3 above)
Comment (nathan-contino-mongo) [reply source/android/link-identities.txt:54]: It's implicit in Kotlin, so in many of the Java examples I've just been using the variable name ``it`` for single parameter lambdas that have a parallel Kotlin example. I do think that result could be more descriptive and helpful, but for the sake of parallelism with most of our other code examples I'll leave this as ``it`` for now. Will refactor all of the Java ``it`` lambda parameters into ``result`` in a future PR -- this is a really good point about helpfulness.

File: source/android/link-identities.txt (hunk 2 above)
Comment (nathan-contino-mongo) [reply source/android/link-identities.txt:29]: it's an object of class ``User`` (capitalized) so I'll leave it that way in text here. You can name your user object whatever you want (though my examples always use ``user`` as the name) so lowercase *could* be slightly confusing to readers.

**Pull request merged**